MSSQL: Panic Not implemented (Grafana v7.5.0)

**What happened**:
```
panic: Not implemented
goroutine 257 [running]:
github.com/denisenkom/go-mssqldb.passthroughConn.SetWriteDeadline(...)
	/root/go/pkg/mod/github.com/denisenkom/go-mssqldb@v0.0.0-20200620013148-b91950f658ec/net.go:167
crypto/tls.(*Conn).SetWriteDeadline(...)
	/usr/local/go/src/crypto/tls/conn.go:150
crypto/tls.(*Conn).closeNotify(0xc000a2fc00, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:1360 +0xfd
crypto/tls.(*Conn).Close(0xc000a2fc00, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:1330 +0xd7
github.com/denisenkom/go-mssqldb.(*Conn).Close(0xc000d940c0, 0x0, 0x0)
	/root/go/pkg/mod/github.com/denisenkom/go-mssqldb@v0.0.0-20200620013148-b91950f658ec/mssql.go:361 +0x3a
database/sql.(*driverConn).finalClose.func2()
	/usr/local/go/src/database/sql/sql.go:592 +0x49
database/sql.withLock(0x2c388b0, 0xc000426990, 0xc0009e9868)
	/usr/local/go/src/database/sql/sql.go:3294 +0x69
database/sql.(*driverConn).finalClose(0xc000426990, 0x0, 0xc0009e98e8)
	/usr/local/go/src/database/sql/sql.go:590 +0x13c
database/sql.(*driverConn).Close(0xc000426990, 0xc000426990, 0x0)
	/usr/local/go/src/database/sql/sql.go:571 +0x135
database/sql.(*DB).putConn(0xc0009632b0, 0xc000426990, 0x0, 0x0, 0x2c38801)
	/usr/local/go/src/database/sql/sql.go:1398 +0x2f7
database/sql.(*driverConn).releaseConn(...)
	/usr/local/go/src/database/sql/sql.go:475
database/sql.(*Rows).close(0xc000b16800, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/database/sql/sql.go:3190 +0x164
database/sql.(*Rows).Close(...)
	/usr/local/go/src/database/sql/sql.go:3161
database/sql.(*Rows).Next(0xc000b16800, 0x3f8d008)
	/usr/local/go/src/database/sql/sql.go:2846 +0xb8
github.com/grafana/grafana/pkg/tsdb/sqleng.(*sqlQueryEndpoint).transformToTable(0xc000213500, 0xc00090e840, 0xc000b725f0, 0xc000afa580, 0xc00090e800, 0x5, 0x0)
	/drone/src/pkg/tsdb/sqleng/sql_engine.go:259 +0x2b1
github.com/grafana/grafana/pkg/tsdb/sqleng.(*sqlQueryEndpoint).Query.func1(0xc0009ef370, 0xc000213500, 0xc00090e800, 0xc000a061b0, 0x82, 0xc00090e840, 0xc000afa580)
	/drone/src/pkg/tsdb/sqleng/sql_engine.go:192 +0x2f9
created by github.com/grafana/grafana/pkg/tsdb/sqleng.(*sqlQueryEndpoint).Query
	/drone/src/pkg/tsdb/sqleng/sql_engine.go:165 +0x45c
```

**What you expected to happen**:
No panic.

**How to reproduce it (as minimally and precisely as possible)**:
Not sure yet.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.5.0
- Data source type & version: Microsoft SQL Server (it seems)
- OS Grafana is installed on: Grafana Cloud

--- a/go.mod
+++ b/go.mod
@@ -4,7 +4,8 @@ go 1.15
 
 // Override xorm's outdated go-mssqldb dependency, since we can't upgrade to current xorm (due to breaking changes).
 // We need a more current go-mssqldb so we get rid of a version of apache/thrift with vulnerabilities.
-replace github.com/denisenkom/go-mssqldb => github.com/denisenkom/go-mssqldb v0.0.0-20200620013148-b91950f658ec
+// Also, use our fork with fixes for unimplemented methods (required for Go 1.16).
+replace github.com/denisenkom/go-mssqldb => github.com/grafana/go-mssqldb v0.0.0-20210326084033-d0ce3c521036
 
 // Override k8s.io/client-go outdated dependency, which is an indirect dependency of grafana/loki.
 // It's also present on grafana/loki's go.mod so we'll need till it gets updated.

